src/resolve/resolved-dns-stream.c

```diff
@@ -183,6 +183,36 @@ static int dns_stream_identify(DnsStream *s) {
         return 0;
 }
 
+static ssize_t dns_stream_writev(DnsStream *s, struct iovec *iov, int iovcnt) {
+        ssize_t r;
+
+        assert(s);
+        assert(iov);
+
+        if (s->tfo_salen) {
+                struct msghdr hdr = {};
+
+                hdr.msg_iov = iov;
+                hdr.msg_iovlen = iovcnt;
+                hdr.msg_name = &s->tfo_address->sa;
+                hdr.msg_namelen = s->tfo_salen;
```

Comment:
please, fold this into the structure initialization above. It's not just nicer to look at but compilers usually generate better code too that way. i.e.:

```c
struct msghdr hdr = {
        .msg_iov = iov,
        .msg_iovlen …
…
}
```